patches/api/0346-Schooling-Fish-API.patch

```diff
@@ -0,0 +1,92 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Tue, 7 Dec 2021 19:50:01 -0500
+Subject: [PATCH] Schooling Fish API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/SchoolingFish.java b/src/main/java/io/papermc/paper/entity/SchoolingFish.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..a2fbefe377157620be09e31496131a50c014e81d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/SchoolingFish.java
+@@ -0,0 +1,47 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.entity.Fish;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Represents a fish that can school with other fish.
++ */
++public interface SchoolingFish extends Fish {
++
++    /**
++     * Forces this fish to follow the given fish.
++     *
++     * @param leader fish to follow
++     */
++    void startFollowing(@NotNull SchoolingFish leader);
++
++    /**
++     * Causes the fish to stop following their current
++     * leader.
++     */
++    void stopFollowing();
++
++    /**
++     * Returns if this fish under normal circumstances can be followed.
++     *
++     * @return can be followed or not
++     */
++    boolean canBeFollowed();
++
++    /**
++     * Returns the amount of fish currently following this fish.
++     *
++     * @return school size
++     */
++    int getSchoolSize();
++
++    /**
++     * Gets the fish that this entity is currently following.
++     *
++     * @return following fish
++     */
++    @Nullable
++    SchoolingFish getCurrentFollowedFish();
```

Comment:
```suggestion
+    SchoolingFish getCurrentlyFollowedFish();
```
english isn't my main lang but this sounds more correct to me?

Reply:
Would 
getSchoolFollowTarget()
fit?